File: Algorithms/C++/1302-Deepest_Leaves_Sum.cpp
// BFS
// Runtime: 48 ms, faster than 73.18% of C++ online submissions for Deepest Leaves Sum.
// Memory Usage: 33.7 MB, less than 100.00% of C++ online submissions for Deepest Leaves Sum.
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int deepestLeavesSum(TreeNode* root) {
        vector<TreeNode* > layer = {root};
        int res = 0;
        while (layer.size()) {
            vector<TreeNode* > next_layer;
            int next_res = 0;
            for (TreeNode* node : layer) {
                next_res += node->val;
                if (node->left)
                    next_layer.push_back(node->left);
                if (node->right)
                    next_layer.push_back(node->right);
            }
            layer = next_layer;
            res = next_res;
        }
        return res;
    }
};

File: Algorithms/Python3.x/1302-Deepest_Leaves_Sum.py
# BFS
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def deepestLeavesSum(self, root: TreeNode) -> int:
        layer = [root]
        res = root.val
        while layer:
            next_layer = []
            next_res = 0
            for node in layer:
                next_res += node.val
                if node.left:
                    next_layer.append(node.left)
                if node.right:
                    next_layer.append(node.right)
            layer = next_layer
            res = next_res
        return res
